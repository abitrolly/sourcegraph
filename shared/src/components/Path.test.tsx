import React from 'react'
import { Path } from './Path'
import { mount } from 'enzyme'

describe('Path', () => {
    test('no path components', () => {
        expect(mount(<Path path="" />)).toMatchSnapshot()
    })

    test('1 path component', () => {
        expect(mount(<Path path="a" />)).toMatchSnapshot()
    })

    test('2 path components', () => {
        expect(mount(<Path path="a/b" />)).toMatchSnapshot()
    })

    test('3 path components', () => {
        expect(mount(<Path path="a/b/c" />)).toMatchSnapshot()
    })
})
