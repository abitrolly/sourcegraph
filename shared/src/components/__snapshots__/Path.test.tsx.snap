// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Path 1 path component 1`] = `
<Path
  path="a"
>
  <strong>
    a
  </strong>
</Path>
`;

exports[`Path 2 path components 1`] = `
<Path
  path="a/b"
>
  <span
    className="text-muted"
  >
    a
    /
  </span>
  <strong>
    b
  </strong>
</Path>
`;

exports[`Path 3 path components 1`] = `
<Path
  path="a/b/c"
>
  <span
    className="text-muted"
  >
    a/b
    /
  </span>
  <strong>
    c
  </strong>
</Path>
`;

exports[`Path no path components 1`] = `
<Path
  path=""
/>
`;
